### PR TITLE
Plugin Extensions should be detected with any ClassLoadingStrategy

### PR DESCRIPTION
**Pull request by etojesaandro**
Always look for the EXTENSIONS_RESOURCE in PluginClassLoader first (PDA strategy)

### CODE REVIEW

**Comment by decebals** [comment]
@etojesaandro 
It would be nice if you could add a test to this PR.
We have [PluginClassLoaderTest](https://github.com/pf4j/pf4j/blob/master/pf4j/src/test/java/org/pf4j/PluginClassLoaderTest.java).

**Comment by etojesaandro** [comment]
> @etojesaandro It would be nice if you could add a test to this PR. We have [PluginClassLoaderTest](https://github.com/pf4j/pf4j/blob/master/pf4j/src/test/java/org/pf4j/PluginClassLoaderTest.java).

The set of tests were implemented

File: pf4j/src/test/java/org/pf4j/PluginClassLoaderTest.java [hunk 1]
```diff
@@ -74,6 +75,8 @@ static void setUpGlobal() throws IOException, URISyntaxException {
         createFile(parentClassPathBase.resolve("META-INF").resolve("file-in-both-parent-and-dependency-and-plugin"));
         createFile(parentClassPathBase.resolve("META-INF").resolve("file-in-both-parent-and-dependency"));
         createFile(parentClassPathBase.resolve("META-INF").resolve("file-in-both-parent-and-plugin"));
+        createFile(parentClassPathBase.resolve(LegacyExtensionStorage.EXTENSIONS_RESOURCE));
+
```
Comment (decebals) [inline pf4j/src/test/java/org/pf4j/PluginClassLoaderTest.java:79]: Remove this empty line

File: pf4j/src/test/java/org/pf4j/PluginClassLoaderTest.java [hunk 2]
```diff
@@ -106,7 +109,8 @@ void setUp() throws IOException {
                 .addFile(Paths.get("classes/META-INF/dependency-file"), "dependency")
                 .addFile(Paths.get("classes/META-INF/file-in-both-parent-and-dependency-and-plugin"), "dependency")
                 .addFile(Paths.get("classes/META-INF/file-in-both-parent-and-dependency"), "dependency")
-                .build();
+                .addFile(Paths.get("classes/" + LegacyExtensionStorage.EXTENSIONS_RESOURCE), "dependency")
+            .build();
```
Comment (decebals) [inline pf4j/src/test/java/org/pf4j/PluginClassLoaderTest.java:113]: Preserve indentation

**Review by decebals — CHANGES_REQUESTED** [review]
It looks good for me. Only some minor requests.

**Comment by decebals** [comment]
@etojesaandro Thanks for your contribution!

**Pull request merged**